enclave_wrangler/dataset_download_new_api.py

```diff
@@ -12,67 +16,161 @@
+            os.mkdir(cache_dir)
+
+        # Request
+        results: List[Dict] = []
+        url_base = f'{self.base_url}/api/v1/ontologies/{self.ontology_rid}/objects/{object_type}'
+        url = url_base
+        while True:
+            response = requests.get(url, headers=self.headers)
+            response_json = response.json()
+            results += response_json['data']
+            if 'nextPageToken' not in response_json or not response_json['nextPageToken']:
+                break
+            url = url_base + '?pageToken=' + response_json["nextPageToken"]
+
+        # Parse / cache
+        # Get rid of nested 'properties' key, and add 'rid' in with the other fields
+        results = [{**x['properties'], **{'rid': x['rid']}} for x in results]
+        df = pd.DataFrame(results).fillna('')
+
+        # TODO: temporary cache code until decide where/how to write these
+        if not os.listdir(cache_dir):
+            df.to_csv(cache_path, index=False)
+
+        return results
+
+    def link_types(self) -> List[Dict]:
+        """Get link types
+        curl -H "Content-type: application/json" -H "Authorization: Bearer $OTHER_TOKEN" \
+        "https://unite.nih.gov/ontology-metadata/api/ontology/linkTypesForObjectTypes" --data '{
+            "objectTypeVersions": {
+                "ri.ontology.main.object-type.a11d04a3-601a-45a9-9bc2-5d0e77dd512e": \
+                "00000001-9834-2acf-8327-ecb491e69b5c"
+            }
+        }'
+        jq '..|objects|.apiName//empty' $@
+        """
+        # TODO: @Siggie I tried using the above curl in Python but I got this (- Joe 2022/08/21):
+        #  {'errorCode': 'INVALID_ARGUMENT', 'errorName': 'Conjure:UnsupportedMediaType', 'errorInstanceId':
+        #  '7976c277-8187-4a5a-91b9-2e8bd1c9934c', 'parameters': {}}
+        data = {
+            "objectTypeVersions": {
+                "ri.ontology.main.object-type.a11d04a3-601a-45a9-9bc2-5d0e77dd512e":
+                    "00000001-9834-2acf-8327-ecb491e69b5c"
+            }
+        }
+        url = f'{self.base_url}/ontology-metadata/api/ontology/linkTypesForObjectTypes'
+        response = requests.post(url, headers=self.headers, data=data)
+        response_json = response.json()
+        return response_json
+
+
+def run(request_types: List[str]) -> Dict[str, Dict]:
     """Run"""
-    types = objTypes()
-    print('\n'.join([t['apiName'] for t in types]))
-
+    client = EnclaveClient()
+    request_funcs: Dict[str, Callable] = {
+        'objects': client.objects,
+        'object_types': client.obj_types,
+        'link_types': client.link_types,
+    }
+    results = {}
+    for req in request_types:
+        results[req] = request_funcs[req]()
+        # if req == 'object_types':
+        #     print('\n'.join([t['apiName'] for t in results[req]['types']]))
+    return results
+
+
+# TODO: @Siggie: Temp; just want to look at all the data before deciding what to do w/ it
+def cache_objects_of_interest():
```

Comment:
@Siggie: Temp; just want to look at all the data before deciding what to do w/ it